calculator.go

```diff
@@ -1,13 +1,97 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+)
+
+// Add takes a list of numbers and returns the result of adding them together.
+func Add(a, b float64, nums ...float64) float64 {
+	result := a + b
+	for _, n := range nums {
+		result += n
+	}
+	return result
+}
+
+// Subtract takes a list of numbers and returns the result of subtracting them from each other.
+func Subtract(a, b float64, nums ...float64) float64 {
+	result := a - b
+	for _, n := range nums {
+		result -= n
+	}
+	return result
+}
+
+// Multiply takes a list of numbers and multiplies them.
+func Multiply(a, b float64, nums ...float64) float64 {
+	result := a * b
+	for _, n := range nums {
+		result *= n
+	}
+	return result
+}
+
+// Divide takes a list of numbers and divides them.
+// Error is returned when dividing by 0.
+func Divide(a, b float64, nums ...float64) (float64, error) {
+	err := errors.New("cannot divide by zero")
+	if a == 0 || b == 0 {
+		return 0, err
+	}
+
+	var result float64 = a / b
+	for _, n := range nums {
+		if n == 0 || result == 0 {
+			return 0, err
+		}
+		result /= n
+	}
+	return result, nil
+}
+
+// Sqrt finds the square root of a number
+// Running this on 0 or negative numbers returns an error
+func Sqrt(num float64) (float64, error) {
+	if num < 0 {
+		return 0, fmt.Errorf("not allowed to find the square root of a negative number: %f", num)
+	}
+	return math.Sqrt(num), nil
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// StringMath takes a string, parses it and runs the math operation found therein
+// Returns an error if it cannot do so
+func StringMath(input string) (float64, error) {
+	var a, b float64
+	var operation string
+	var err error
+
+	operations := []string{"+", "-", "*", "/"}
+	for _, op := range operations {
+		if strings.Contains(input, op) {
+			operation = op
+			input = strings.Replace(input, op, " ", 1)
+			_, err = fmt.Sscanf(input, "%f%f", &a, &b)
```

Comment:
We could just call `fmt.Sscanf` directly, couldn't we, without using `strings.Contains` first.

Reply:
I must admit this one got me stuck for a while, even the first time you suggested `fmt.Sscanf`.
I would love to do everything by calling it directly without any machinery.

The issue is that Sscanf will only work with space-separated values, which is the reason I went with the machinery that replaces whatever operator is found in the string (line 74) with a whitespace (line 76).

Without this, the tests for string values like `"2*2"` fail.
```
var a, b float64
var op string
operation := "2*2"
	
fmt.Sscanf(operation, "%f%s%f", &a, &op, &b)
fmt.Println(a, op, b) // 2 *2 0
```

A bit stumped here :)

Reply:
Well, the problem here is that Scanf is being too greedy, isn't it? You asked it to find a string starting at `*` and it did, but it went on to see how much more of a string it could read, and it read as far as `*2`. That leaves nothing for the second value.

Perhaps we could tell it to be less greedy? Is there a way to specify in the format string that we're only interested in a string of exactly one character for the operator? See what you can find out.

Reply:
😂 Thanks for that. Addressed in https://github.com/micuffaro/ftl-fundamentals/pull/1/commits/e8408efc091c65add548f0c73fdbc6201feac7a1